proposals/reading_list.md

```diff
@@ -0,0 +1,178 @@
+# New API: browser.readingList()
+
+## Background
+
+Chromium has a reading list feature which allows users to store websites / media they would like to consume or keep reference to in the future.
+
+On Chromium this feature supports the following operations:
+- Adding an entry to the ReadingList
+- Remove an entry from the ReadingList
+- Marking an entry as read / unread
+
+At present, extensions developers are unable to read from and modify the reading list feature in the browser.
+
+## Problem
+
+The reading list feature on Chromium does not have an existing API which allows extensions developers the ability to read from or modify this feature. Additionally, the UI allotted to extensions within the browser is much smaller compared to that of the reading list and similar features housed in the browsers side panel.
+
+## Solution
+
+Considering the above situation, we'd like to propose a new extension API, `browser.readingList`, which will allow developers to read from and modify the reading list in any supported browser.
+
+# API Proposal
+
+The `browser.readingList` will introduce a few new types and functions. Below is the schema in the IDL format:
+
+## Types
+A ReadingListEntry is the extensions representation of a Reading List object. Where every URL stored in the Reading List must be unique.
+```js
+ dictionary ReadingListEntry {
+	// The url of the entry.
+	DOMString url;
```

Comment:
Could we say a bit more about what we mean by URL here? Can it have query parameters? Can you have two entries with the same hostname and path but a different hash (for example)?

Reply:
Oh I see now! 

Yes, the URL can have query parameters. Two entries with the same hostname and path but different hashes are also allowed. As long as the string of the url is different even by a single letter. I will update the comment to reflect this. 

Another thing is that the URL must be "valid". For chromium, this means `chrome://` paths cannot be added in addition to any URLs that do not have protocols or hostnames.

Reply:
Thank you!